mgmtd/mgmt_main.c

```diff
@@ -143,8 +143,16 @@ static struct frr_signal_t mgmt_signals[] = {
 extern const struct frr_yang_module_info frr_staticd_cli_info;
 #endif
 
+#if HAVE_BFDD == 0 || defined(HAVE_RTADV)
```

Comment:
using 2 different check methods here, how about just use `defined(HAVE_...)` and `!defined(HAVE_...)` rather than `== 0`

Reply:
`HAVE_BFDD` is always defined to either 0 or 1. I have no idea why it was implemented this way, but I'm not going to rework it in this PR, so we can't use `defined` here.

For the second question, yes, I think your suggestion is better. I actually don't know why didn't I implement it like this in the first place. Will fix in a cleanup commit/PR.

Reply:
Actually libyang accepts `{NULL}` as well, so it's even simpler:
```
const char *features[] = {
#if HAVE_BFDD == 0
	"ptm-bfd",
#endif
#if defined(HAVE_RTADV)
	"ipv6-router-advertisements",
#endif
	NULL
};
```
Without this wrapping condition.